scanner/decisions/0004-scannerv4-read-vulneralbilities.md

```diff
@@ -0,0 +1,24 @@
+# 0004 - Scanner V4 NVD CVSS enrichment data updating pipeline
+
+- **Author(s):** Yi Li
+- **Created:** [2023-09-11]
+
+## Context
+
+We are in the process of transitioning the StackRox Scanner to utilize ClairCore as its primary scanning engine. Our objective is twofold: firstly, to allow Central to fetch NVD CVSS data, and secondly, to ensure the Scanner V4 seamlessly connecting with Central, triggering data updates and enriching vulnerabilities with CVSS data. Throughout this pipeline, the Claircore enricher will play a pivotal role. It will manage tasks including downloading CVSS data from Google storage, parsing the information, and updating the data accordingly for Scanner V4
+
+## Decision
+
+The CVSS Updater resides within Central, fetching CVSS data from Google storage upon request. This updater then stores the data in Central's file system. Throughout this download and save process, the Claircore CVSS enricher plays a significant role.
```

Comment:
> The CVSS Updater resides within Central, fetching CVSS data from Google storage upon request.

Perhaps it might be worth considering having it pull updates from the Storage on a regular basis, rather than upon request. This would allow us to respond with a "not modified" response in the CVSS Updater endpoint in Central, and eliminate the need to contact the storage every time. I can see how this approach could be particularly helpful, especially since updates to the storage content don't occur frequently.